### PR TITLE
[Issue #2351, #2472] Migrate to platforms step function pattern

### PR DESCRIPTION
**Pull request by coilysiren**
## Summary

Relates to: #2351, #2472

## Changes proposed

There are two goals to this PR:

1. Make a material change that gets us closer in line with the platform. Specifically we are pulling in https://github.com/navapbc/template-infra/pull/745
2. Updates the step functions in advance of adding another step function - for the terraform in ECS stuff

## Context for reviewers

The majority of the diff is security related, specifically: 

- via guidance from @lorenyu, the permissions for managing the step function and workflow triggers were moved off of the task exec role. that's the diff on `infra/modules/service/access-control.tf`
- those permissions were moved to `infra/modules/service/scheduler_role.tf` and `infra/modules/service/workflow_orchestrator_role.tf`

The rest of the diff is around easy of configuration:

- `infra/modules/service/scheduled_jobs.tf` represents the new, highly configurable, way to create scheduled jobs
- `infra/api/app-config/env-config/scheduled_jobs.tf` shows how two scheduled jobs are configured

## Additional information

image

image

### CODE REVIEW

File: infra/api/app-config/env-config/scheduled_jobs.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      "--load",
+      "--transform",
+      "--set-current",
+    ],
+  }
+  scheduled_jobs = {
+    copy-oracle-data = {
+      task_command        = ["poetry", "run", "flask", "data-migration", "copy-oracle-data"]
+      schedule_expression = "rate(2 minutes)"
+    }
```
Comment (coilysiren) [inline infra/api/app-config/env-config/scheduled_jobs.tf:45]: @chouinar fancy new simple way to setup scheduled jobs

File: infra/api/app-config/env-config/scheduled_jobs.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      "load-transform",
+      "--load",
+      "--transform",
+      "--set-current",
+    ],
+  }
+  scheduled_jobs = {
+    copy-oracle-data = {
+      task_command        = ["poetry", "run", "flask", "data-migration", "copy-oracle-data"]
+      schedule_expression = "rate(2 minutes)"
```
Comment (mdragon) [inline infra/api/app-config/env-config/scheduled_jobs.tf:44]: Should we include enabled/disabled here too, or does a null schedule or something like that cause the job to just not run? Just thinking there may be times we want to keep the job from running, but not delete it from here entirely.

File: infra/modules/service/scheduled_jobs.tf [hunk 3]
```diff
@@ -0,0 +1,86 @@
+resource "aws_scheduler_schedule" "scheduled_jobs" {
+  for_each = var.scheduled_jobs
+
+  # TODO(https://github.com/navapbc/template-infra/issues/164) Encrypt with customer managed KMS key
+  # checkov:skip=CKV_AWS_297:Encrypt with customer key in future work
+
+  name                         = "${var.service_name}-${each.key}"
+  state                        = "ENABLED"
+  schedule_expression          = each.value.schedule_expression
```
Comment (mdragon) [inline infra/modules/service/scheduled_jobs.tf:9]: Noting the enabled here that we might want to control via the list of jobs per the other comment.

File: infra/modules/service/scheduled_jobs.tf [hunk 4]
```diff
@@ -0,0 +1,86 @@
+resource "aws_scheduler_schedule" "scheduled_jobs" {
+  for_each = var.scheduled_jobs
+
+  # TODO(https://github.com/navapbc/template-infra/issues/164) Encrypt with customer managed KMS key
+  # checkov:skip=CKV_AWS_297:Encrypt with customer key in future work
+
+  name                         = "${var.service_name}-${each.key}"
+  state                        = "ENABLED"
```
Comment (mdragon) [inline infra/modules/service/scheduled_jobs.tf:8]: Noting the enabled here that we might want to control via the list of jobs per the other comment.

**Review by mdragon — APPROVED** [review]
Approving as is. If we do want to facilitate pausing jobs via config as suggested in the comments, I'm happy to have that come in a future ticket.

File: infra/api/app-config/env-config/scheduled_jobs.tf (hunk 1 above)
Comment (chouinar) [reply infra/api/app-config/env-config/scheduled_jobs.tf:45]: Thanks, this is simple enough that I can set things up / not bother you for any adjustments we make.

File: infra/api/app-config/env-config/scheduled_jobs.tf (hunk 2 above)
Comment (coilysiren) [reply infra/api/app-config/env-config/scheduled_jobs.tf:45]: Added! https://github.com/HHS/simpler-grants-gov/pull/2506/commits/1ba3d0615fdff173d9020b34a7e7fc83cfe206e9

**Review by mdragon — APPROVED** [review]

**Pull request merged**